src/ansi-c/expr2c.cpp

```diff
@@ -378,7 +383,10 @@ std::string expr2ct::convert_rec(
 
     dest+=" }";
 
-    dest+=d;
+    if(src.find(ID_typedef).is_not_nil())
+      dest+=src.get_string(ID_typedef);
+    else
+      dest+=d;
```

Comment:
@peterschrammel How come we only add on the `d` (declarator) string if not a typedef. Not really clear on exactly what could be in `d` at this point.